rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -332,11 +333,14 @@ impl LightFetch {
 			// match them with their numbers for easy sorting later.
 			let bit_combos = filter.bloom_possibilities();
 			let receipts_futures: Vec<_> = self.client.ancestry_iter(filter.to_block)
-				.take_while(|ref hdr| BlockId::Number(hdr.number()) != filter.from_block)
-				.take_while(|ref hdr| BlockId::Hash(hdr.hash()) != filter.from_block)
+				.take_while(|ref hdr|{
+					BlockId::Hash(hdr.hash()) != filter.from_block
+					&& BlockId::Number(hdr.number()) != filter.from_block
+				})
+				.chain(Some(self.client.block_header(filter.from_block).expect("checked before")))
```

Comment:
`block_header` can return `None` and I don't see where do we check it's not `None`, could you point me to that line?

Reply:
It was tested in the case of a block hash (in `block_number` ) and I added a condition before to make sure the number case would be in range to, it is certainly not explicit enough.
I did a little change to test it explicitly in my last commit (plus remove the need to have two condition in takewhile).